### PR TITLE
lint: disallow specific global variables such as event, error.

### PR DESCRIPTION
**Pull request by antoine-amara**
add a new lint rule: `restricted-globals`

disallow the use of some js global variables such as events and errors,
as this is considered as a bad practice.

> Note: for now, only `event` and `error` globals are disallowed

### CODE REVIEW

File: packages/@romejs/diagnostics/descriptions.ts [hunk 1]
```diff
@@ -458,6 +458,10 @@ export const descriptions = createMessages(
             ],
           };
       },
+      RESTRICTED_GLOBALS: (globalName) => ({
+        category: 'lint/restrictedGlobals',
+        message: `don't use global variable ${globalName}, use local variable instead`,
```
Comment (sebmck) [inline packages/@romejs/diagnostics/descriptions.ts:463]: This looks great! Just a suggestion on the diagnostic message:

```suggestion
        message: markup`<emphasis>${globalName}</emphasis> isn't defined locally but is a valid global variable`,
```

Comment (antoine-amara) [reply packages/@romejs/diagnostics/descriptions.ts:463]: this new diagnostic message is a little bit confusing for me. We don't inform the user the global variable use is disallowed. I try to change the diagnostic message to: `the use of the existing global variable <emphasis>${globalName}</emphasis> is not allowed. Use local parameter instead.`. Say to me what do you think about this one.

Comment (sebmck) [reply packages/@romejs/diagnostics/descriptions.ts]: Few nits: It's not capitalized. "Parameter" isn't broad enough, a parameter is specifically a function parameter, we should call it a binding/variable.

Comment (antoine-amara) [reply packages/@romejs/diagnostics/descriptions.ts]: thanks for the review, I push some corrections.

**Comment by antoine-amara** [comment]
@sebmck, I rebase the branch, update the unit tests and the diagnostic message.

**Comment by sebmck** [comment]
Looks great, thank you! Sorry for the delay in reviewing/merging.

**Pull request merged**